Meta/Lagom/CMakeLists.txt

```diff
@@ -105,59 +109,51 @@ include_directories(${CMAKE_BINARY_DIR})
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
 # install rules, think about moving to its own helper cmake file
-# Don't install Lagom libs into the target Root/
-# FIXME: Remove this check for 4594
-if (CMAKE_SOURCE_DIR MATCHES ".*/Lagom")
-    include(CMakePackageConfigHelpers)
-    include(GNUInstallDirs)
-
-    # find_package(<package>) call for consumers to find this project
-    set(package Lagom)
-
-    write_basic_package_version_file(
-        "${package}ConfigVersion.cmake"
-        COMPATIBILITY SameMajorVersion
-    )
+include(CMakePackageConfigHelpers)
+include(GNUInstallDirs)
 
-    # Allow package maintainers to freely override the path for the configs
-    set(Lagom_INSTALL_CMAKEDIR "${CMAKE_INSTALL_DATADIR}/${package}"
-        CACHE PATH "CMake package config location relative to the install prefix")
-    mark_as_advanced(Lagom_INSTALL_CMAKEDIR)
+# find_package(<package>) call for consumers to find this project
+set(package Lagom)
 
-    install(
-        FILES ${SERENITY_PROJECT_ROOT}/Meta/CMake/lagom-install-config.cmake
-        DESTINATION "${Lagom_INSTALL_CMAKEDIR}"
-        RENAME "${package}Config.cmake"
-        COMPONENT Lagom_Development
-    )
+# Allow package maintainers to freely override the path for the configs
+set(Lagom_INSTALL_CMAKEDIR "${CMAKE_INSTALL_DATADIR}/${package}"
+    CACHE PATH "CMake package config location relative to the install prefix")
+mark_as_advanced(Lagom_INSTALL_CMAKEDIR)
 
-    install(
-        FILES "${PROJECT_BINARY_DIR}/${package}ConfigVersion.cmake"
-        DESTINATION "${Lagom_INSTALL_CMAKEDIR}"
-        COMPONENT Lagom_Development
-    )
+write_basic_package_version_file(
+    "${package}ConfigVersion.cmake"
+    COMPATIBILITY SameMajorVersion
+    ARCH_INDEPENDENT # Because Lagom is used for host tools, ignore arch check for when we're cross-compiling
+)
 
-    install(
-        EXPORT LagomTargets
-        NAMESPACE Lagom::
-        DESTINATION "${Lagom_INSTALL_CMAKEDIR}"
-        COMPONENT Lagom_Development
-    )
+configure_package_config_file(
```

Comment:
`configure_package_config_file` is more of a utility function for those projects that have a little more advanced config file with package components and other needs. As it stands now, the config file is just a proxy to the exported target set, so reaching for this function is unnecessary. It also depends on you providing the install path at configure time, but that can be changed at install time:

```sh
cmake .. -D CMAKE_INSTALL_PREFIX=foo # install to foo
cmake --build .
cmake --install . --prefix bar # actually no, install to bar
```

Reply:
Gotcha. I had added that @PACKAGE_INIT@ and configure_package_config_file when trying to debug why things weren't working and grasping for straws. Turns out the solution to why it wasn't finding the targets was that the host build was x86_64 and the target build was i686 and I needed to specify ARCH_INDEPENDENT on the package version file creator. So no qualms in removing it.

Reply:
Technically you don't even need versioning either, since (seemingly) nothing else is versioned in SerenityOS. That's a good reminder. I don't see a scenario where `find_package(Lagom <version>)` would be useful.

Reply:
Eh, true, the policy of the project is very much Live-At-Head and Live-In-Tree. I'm inclined to leave it under the assumption that CMake will get upset if it's not present?

Reply:
No, the version config is completely optional. The only time CMake ever looks for it is when a version is requested and when it's not found then CMake will simply consider that particular package unsitable and move on to the next one.

[Docs](https://cmake.org/cmake/help/latest/command/find_package.html#version-selection):

> If no such version file is available then the configuration file is assumed to not be compatible with any requested version.